src/java/org/apache/cassandra/db/repair/CassandraTableRepairManager.java

```diff
@@ -118,6 +118,8 @@ public boolean apply(SSTableReader sstable)
         }
         catch (Exception ex)
         {
+            if (ex instanceof InterruptedException)
```

Comment:
if that's a bug we should fix it upstream (`trunk` still doesn't have `interrupt()` here)